consensus/src/metrics.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use prometheus::{default_registry, register_int_gauge_vec_with_registry, IntGaugeVec, Registry};
+
+#[derive(Clone, Debug)]
+pub struct ConsensusMetrics {
+    /// The number of elements in the Dag (for Tusk or Bullshark)
+    pub consensus_dag_size: IntGaugeVec,
+    /// The last committed round from consensus
+    pub last_committed_round: IntGaugeVec,
+    /// The number of elements from the vertices secondary index (external consensus)
+    pub external_consensus_dag_vertices_elements: IntGaugeVec,
+    /// The number of elements in the dag (external consensus)
+    pub external_consensus_dag_size: IntGaugeVec,
+}
+
+impl ConsensusMetrics {
+    pub fn new(registry: &Registry) -> Self {
+        Self {
+            consensus_dag_size: register_int_gauge_vec_with_registry!(
```

Comment:
Sorry I'm late to the party.

The reason to use a `vec` is to add labels so you can have different time series, for example it could be for different workers, or something.  If you aren't using any labels, then instead you should just use the simple `int_gauge`, not one with a vec.

Reply:
Ah thanks for this , I am going to have a look on it.